dimod/reference/composites/scalecomposite.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2018 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+"""
+A composite that fixes the variables provided and removes them from the
+bqm object before sending to its child sampler.
+
+See `Ocean Glossary <https://docs.ocean.dwavesys.com/en/latest/glossary.html>`_ for explanations
+of technical terms in descriptions of Ocean tools.
+"""
+try:
+    import collections.abc as abc
+except ImportError:
+    import collections as abc
+
+from numbers import Number
+
+import numpy as np
+
+from dimod.binary_quadratic_model import BinaryQuadraticModel
+from dimod.core.composite import ComposedSampler
+from dimod.higherorder import poly_energies, _relabeled_poly
+
+__all__ = ['ScaleComposite']
+
+
+class ScaleComposite(ComposedSampler):
+    """Composite to scale variables of a problem
+
+    Inherits from :class:`dimod.ComposedSampler`.
+
+    Scales the variables of a bqm and modifies linear and quadratic terms
+    accordingly.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler
+
+    Examples:
+       This example uses :class:`.ScaleComposite` to instantiate a
+       composed sampler that submits a simple Ising problem to a sampler.
+       The composed sampler scales linear, quadratic biases and offset as
+       indicated by options.
+
+       >>> linear = {'a': -4.0, 'b': -4.0}
+       >>> quadratic = {('a', 'b'): 3.2}
+       >>> sampler = dimod.ScaleComposite(dimod.SimulatedAnnealingSampler())
+       >>> response = sampler.sample_ising(linear, quadratic, scalar=0.5,
+       ...                ignored_interactions=[('a','b')],num_reads = 100)
+       >>> print(response.aggregate().record) # doctest: +SKIP
+       >>> response = sampler.sample_ising(linear, quadratic, scalar=0.5,
+       ...                                              num_reads = 100)
+       >>> print(response.aggregate().record) # doctest: +SKIP
```

Comment:
We're about to make new pretty printing (see #370 ), so maybe just remove most of this example. I would also use `ExactSolver` because the return is deterministic.

```
       >>> linear = {'a': -4.0, 'b': -4.0}
       >>> quadratic = {('a', 'b'): 3.2}
       >>> sampler = dimod.ScaleComposite(dimod.ExactSolver())
       >>> sampleset = sampler.sample_ising(linear, quadratic, scalar=0.5,
       ...                ignored_interactions=[('a','b')],num_reads = 100)
```

Reply:
I cannot use exactsolver as an example for scalecomposite since scaling changes the probabilities. The outputs would look exactly the same and the examples would be meaningless.

Reply:
Fair point, might just be better to skip showing the output then